code-challenges.test.js

```diff
@@ -13,45 +14,181 @@
+// Input: array Output: array
+// Solution: 
+// how to test a function that output is random?
 
 var colors1 = ["purple", "blue", "green", "yellow", "pink"]
 // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
 var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
-
+describe("The shuffles function.", () =>{
+    test("takes in an array, removes the 1st item and shuffles the remaining.", ()=>{
+        expect(shuffles(colors1)).toEqual([ 'blue', 'pink', 'green', 'yellow' ])
+        expect(shuffles(colors2)).toEqual([ 'indigo', 'saffron', 'periwinkle', 'ochre', 'aquamarine' ])
+    })
+})
 
 // b) Create the function that makes the test pass.
-
-
-
-// --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
-
+// Function: shuffles()
+// Input and output: array
+
+// Solution 1: 
+// With random output
+// .shift() method removes the first element of an array and .shift() method changes the original array
+// randomize shuffle-->index-->random pick element, swap 
+const randomShuffles = (arr) =>{
+    // Step 1: .shift(1) remove the first element and get remaining array 
+    arr.shift(1)
+    // Step 2: set up currentIndex to the max number, and will get randomIndex(syntax review: Math.floor(Math.random()*max))
+    let currentIndex = arr.length
+    while (currentIndex >0){
+        //Pick a random element
+        let randomIndex = Math.floor(Math.random()* currentIndex)
+        currentIndex --
+        //Swap with the currentIndex element
+        [arr[currentIndex], arr[randomIndex]] = [arr[randomIndex], arr[currentIndex]]
+    }
+    return arr
+}
+console.log(randomShuffles([1, 5, 90098, "lilj"]))
+// Output1: [ 'lilj', 5, 90098 ]
+// Output2: [ 5, 'lilj', 90098 ]
+// Output3: [ 90098, 'lilj', 5 ]
+
+// Solution 2: 
+// Without random output, shuffled with some rules. arr[currentIndex] will swap with arr[currentIndex - 1]
+const shuffles = (arr) =>{
+    // Step 1: .shift(1) remove the first element and get remaining array 
+    arr.shift(1)
+    // Step 2: set up currentIndex to the max number
+    let currentIndex = arr.length - 1
+    // Start from 1, do until arr[1] swap with arr[0]
+    while (currentIndex > 1){
+        // Pick the element next to the currentIndex and swap
+        [arr[currentIndex], arr[currentIndex-1]] = [arr[currentIndex-1], arr[currentIndex]]
+        currentIndex --
+    }
+    return arr
+}
+console.log(shuffles([1, 5, 90098, "lilj"]))
+// Output: [ 5, 'lilj', 90098 ]
+// console.log(shuffles(colors1))
+// Output: [ 'blue', 'pink', 'green', 'yellow' ]
+// console.log(shuffles(colors2))
+// Output: [ 'indigo', 'saffron', 'periwinkle', 'ochre', 'aquamarine' ]
+
```

Comment:
This is good but when we're writing tests for randomization we're going to want to incorporate those results for each of our tests!